fastlane/lib/fastlane/setup/setup_ios.rb

```diff
@@ -70,6 +73,29 @@ def try_manual_setup
       handle_exception(exception: ex)
     end
 
+    # React Native specific code
+    # Make it easy for people to onboard
+    def react_native_pre_checks
+      return unless self.class.project_uses_react_native?
+      if app_identifier.to_s.length == 0
+        error_message = []
+        error_message << "Could not detect bundle identifier of your react-native app."
+        error_message << "Make sure to open the Xcode project and update the bundle identifier"
+        error_message << "in the `General` section of your project settings."
+        error_message << "Restart `fastlane init` once you're done!"
+        UI.user_error!(error_message.join(" "))
+      end
+    end
+
+    def self.project_uses_react_native?(path: Dir.pwd)
+      package_json = File.join(path, "..", "package.json")
+      return unless File.basename(path) == "ios"
+      return unless File.exist?(package_json)
```

Comment:
I think it would be better if this method always returned `true` or `false`, instead of sometimes returning `nil`, for these checks. :+1: